recipes-core/libhybris/libhybris/0001-Report-support-for-only-2.0-instead-of-3.0.patch

```diff
@@ -0,0 +1,38 @@
+From 8b8d271db718866a777ab9d5876a9c42455856cb Mon Sep 17 00:00:00 2001
+From: MagneFire <IDaNLContact@gmail.com>
+Date: Sat, 1 May 2021 15:08:17 +0200
+Subject: [PATCH] Report support for only 2.0 instead of 3.0. It looks like we
+ experience glitches when supporting 3.0.
+
+---
+ hybris/glesv2/glesv2.c | 14 +++++++++++++-
+ 1 file changed, 13 insertions(+), 1 deletion(-)
+
+diff --git a/hybris/glesv2/glesv2.c b/hybris/glesv2/glesv2.c
+index 716bbde..ed7f366 100644
+--- a/hybris/glesv2/glesv2.c
++++ b/hybris/glesv2/glesv2.c
+@@ -105,7 +105,19 @@ HYBRIS_IMPLEMENT_VOID_FUNCTION3(glesv2, glGetShaderiv, GLuint, GLenum, GLint *);
+ HYBRIS_IMPLEMENT_VOID_FUNCTION4(glesv2, glGetShaderInfoLog, GLuint, GLsizei, GLsizei *, GLchar *);
+ HYBRIS_IMPLEMENT_VOID_FUNCTION4(glesv2, glGetShaderPrecisionFormat, GLenum, GLenum, GLint *, GLint *);
+ HYBRIS_IMPLEMENT_VOID_FUNCTION4(glesv2, glGetShaderSource, GLuint, GLsizei, GLsizei *, GLchar *);
+-HYBRIS_IMPLEMENT_FUNCTION1(glesv2, const GLubyte *, glGetString, GLenum);
++const GLubyte* glGetString (GLenum n1)
++{
++        static const GLubyte * (*f)(GLenum) FP_ATTRIB = NULL;
++        HYBRIS_DLSYSM(glesv2, &f, "glGetString");
++       // Some platforms have glitchy 3.0 driver. Report only 2.0 support.
++	if (n1 == GL_VERSION) {
++		static GLubyte glGetString_versionString[64];
++		snprintf((char *)glGetString_versionString, sizeof(glGetString_versionString), "OpenGL ES 2.0 (%s)", f(n1));
```

Comment:
So I see you call glGetString anyway, is the output  going to be something like "OpenGL ES 2.0 (OpenGL ES 3.0)" ? Ultimately as long as Qt interprets that as "use GLES2" it's fine by me but I'm just a bit surprised :)

Reply:
This implementation is a slightly altered version of: https://github.com/libhybris/libhybris/commit/0ceb7ef5535331e2281a12351145f52787247bdb#diff-0319caa8848d1a067fe51a451b876088522b4068a3cfa483b2f49bc51c6d91c0L683 Click Load diff for `hybris/glesv2/glesv2.c`.

I can also confirm that Qt interprets this fine: https://github.com/qt/qtbase/blob/5.15.2/src/gui/kernel/qplatformopenglcontext.cpp#L137

I did not specifically look if Qt uses GLES2 but it solved the blinking issue that is described in the PR :smile: